.github/workflows/wheel_win_x86_64.yml

```diff
@@ -149,6 +149,9 @@ jobs:
           CIBW_BEFORE_BUILD: |
             pip install pybind11 cmake~=3.24.0 build
 
+          CIBW_ENVIRONMENT: |
+            CMAKE_ARGS="-DPython_EXECUTABLE=$(which python)"
```

Comment:
Since this is Windows, we shouldn't use Linux specific syntax in commands if we can opt for pwsh versions instead.

Reply:
Will find replacement.

Reply:
In my powershell the following works
```
Set-Variable -Name CMAKE_ARGS -Value "-DPython_EXECUTABLE=$(@(where.exe <file pattern> 2>$null)[0])"
```
but it fails here because `where` [does not exists](https://github.com/PennyLaneAI/pennylane-lightning/actions/runs/5059095262/jobs/9080155441?pr=450#step:7:62). By Linux specific syntax, were you thinking `VAR=VAL` or `which`?

Reply:
What about something like capturing in an earlier step and reusing with
```pwsh
$py_exe = Get-Command python | Select-Object -ExpandProperty Definition
```

Or just embedding that line into the string builder:
```pwsh
CMAKE_ARGS="-DPython_EXECUTABLE=$(Get-Command python | Select-Object -ExpandProperty Definition)"
```

Though, if the executable isn't explicitly needed, we can probably drop it.

Reply:
If I just remove that, as on Linux and MacOS, the build passes, but I don't think we target the same Python. If we specify it, we have [paths to Python 3.7](https://github.com/PennyLaneAI/pennylane-lightning/actions/runs/5059416984/jobs/9080925170#step:7:669) and otherwise we get [paths to Python 3.11](https://github.com/PennyLaneAI/pennylane-lightning/actions/runs/5060342821/jobs/9083098290?pr=450#step:7:668). If we look at another branch, say [here](https://github.com/PennyLaneAI/pennylane-lightning/actions/runs/5058922324/jobs/9079823976?pr=452#step:7:664), we have something similar to the later, but we have a path in `C:\Users\runneradmin\AppData` instead of `C:/hostedtoolcache/windows`.  Is that ok? Just not familiar with these Windows envs.

Reply:
We definitely don't want 3.7 for the 3.11 build.

So, if we specify `CMAKE_ARGS="-DPython_EXECUTABLE=$(Get-Command python | Select-Object -ExpandProperty Definition)"` we get 3.11, and if we do not, we get 3.7, or did I misunderstand? I'd be less concerned about the paths, as long as 3.11 is used.

Reply:
No sorry. So 
```
CMAKE_ARGS="-DPython_EXECUTABLE=$(Get-Command python | Select-Object -ExpandProperty Definition)"
```
[just failed](https://github.com/PennyLaneAI/pennylane-lightning/actions/runs/5060148817/jobs/9082661374#step:7:146). But removing `which` was a good call, because even though the build worked out, CMake reports things like 
```
Found PythonLibs: C:/hostedtoolcache/windows/Python/3.7.9/x64/libs/python37.lib
```
Simply removing `-DPythonEXECUTABLE` altogether, CMake resolves 
```
Found Python: C:/hostedtoolcache/windows/Python/3.11.3/x64/python3.exe
```
and this work too.

Reply:
Ah, got it. In this case we can simply ignore the statement and keep the 3.11 builder, since that is appropriately picked up.